src/code/sys_initial_check.c

```diff
@@ -1,17 +1,114 @@
+/**
+ * File: sys_initial_check.c
+ * Description: Functions for checking for the presence of the Expansion Pak and the correct TV type (PAL/NTSC/MPAL),
+ * and functions for printing error messages directly to screen if not found or incorrect.
+ *
+ * These checks are some of the first functions run in Main, before even setting up the system heap, and any image files
+ * are DMA'd directly to fixed RAM addresses.
+ */
 #include "global.h"
+#include "misc/locerrmsg/locerrmsg.h"
+#include "misc/memerrmsg/memerrmsg.h"
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/sys_initial_check/Check_WriteRGBA16Pixel.s")
+// Address with enough room after to load either of the error message image files before the fault screen buffer at the
+// end of RDRAM
+#define CHECK_ERRMSG_STATIC_SEGMENT \
+    (u8*)(FAULT_FB_ADDRESS - sizeof(gExpansionPakNotInstalledErrorTex) - sizeof(gSeeInstructionBookletErrorTex))
```

Comment:
I can't really think of a way to make this friendlier towards modifications of these files..
My first thought was `MAX(sizeof(...) + ..., sizeof(...))` for both files, to ensure the larger of the two is always chosen, but even that isn't great since if either file were modified more `sizeof`s would need adding to the expression anyway. My next thought would be to have ZAPD output a `SIZEOF_<FILE>` define, consisting of the sum of the `sizeof`s of the file's contents, though I'm unsure how much use such a thing would get.
Anyway, something to think about. I don't think waiting on a solution to this is important right now.

Reply:
I think we're just going to have to use a define for the filesize and ifdef for PAL or something: they should have used the segment size, but didn't.